### ISSUE TITLE
False positive in the `test-ended` rule when `t.end` is passed as an argument 

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-ava#60]
The following results in a violation, but it's a common pattern, so I think we should support it.

```js
import test from 'ava';

test.cb(t => {
	setTimeout(t.end, 10);
});
```

**Comment by jfmengels** [comment]
Is `setTimeout` a special case, or should we simply accept any tests that pass `t.end` as an argument?
```js
test.cb(t => {
    someFnWithCb(1, 2, t.end);
});
```

**Comment by jamestalmage** [comment]
I would say any access of `t.end`  (argument or no)
